baselines/jft/al_utils.py

```diff
@@ -16,86 +16,48 @@
 """Utils for active learning datasets."""
 import dataclasses
 import logging
-import re
-from typing import Dict, Iterable, List, Optional, Set, Union
+from typing import Any, Iterable, Optional, Set, Union
 
-from clu.deterministic_data import DatasetBuilder
 import jax
 import tensorflow as tf
 import tensorflow_datasets as tfds
 
 
-# adapted from https://www.tensorflow.org/datasets/determinism
-class TFDSIdToInt:
-  """Format TFDS ids to int."""
-  tfds_id_parser_re = re.compile(r'\w+-(\w+).\w+-(\d+)-of-\d+__(\d+)')
-
-  shard_offsets: Dict[str, List[int]]
-
-  def __init__(self, dataset_info: tfds.core.DatasetInfo, *,
-               splitwise_id: bool):
-    shard_offsets = {}
-    current_offset = 0
-
-    # Build a unique offset across or within splits depending on `splitwise_id`.
-    for split, split_info in sorted(dataset_info.splits.items()):
-      if splitwise_id:
-        current_offset = 0
-
-      split_shard_offsets = []
-      for shard_length in split_info.shard_lengths:
-        current_length = current_offset + shard_length
-        split_shard_offsets.append(current_offset)
-        current_offset = current_length
-
-      shard_offsets[split] = split_shard_offsets
-
-    self.shard_offsets = shard_offsets
-
-  def __call__(self, tfds_id: str) -> int:
-    """Format the tfds_id in a more human-readable."""
-    match = self.tfds_id_parser_re.match(tfds_id)
-    split_name, shard_id, ex_id = match.groups()
-    shard_offsets = self.shard_offsets[split_name]
-    return shard_offsets[int(shard_id)] + int(ex_id)
-
-
 def _subset_generator(*, dataset: tf.data.Dataset,
-                      dataset_info: tfds.core.DatasetInfo,
-                      subset_ids: Optional[Set[int]], splitwise_id: bool):
+                      subset_ids: Optional[Set[int]]):
   """Create a subset based on ids."""
-  tfds_id_to_int = TFDSIdToInt(dataset_info, splitwise_id=splitwise_id)
 
   def inner():
     if subset_ids is not None:
       # Hard fail on type errors
       assert all(map(lambda id: isinstance(id, int), subset_ids))
 
-    for record in dataset:
-      tfds_id = record['tfds_id'].numpy().decode('utf-8')
-      int_id = tfds_id_to_int(tfds_id)
+    for int_id, record in enumerate(dataset):
```

Comment:
Cache tfds ids in a dict and ensure consistency for the training set (when subset_ids is not None).

This is cheaper than caching everything and will still most likely catch any bugs.

Reply:
Sounds good. Once the tests are passing, I'll add this.